crates/bevy_ecs/src/system/system_param.rs

```diff
@@ -136,8 +136,8 @@ impl<'w, 's, Q: WorldQuery + 'static, F: ReadOnlyWorldQuery + 'static> SystemPar
 }
 
 // SAFETY: QueryState is constrained to read-only fetches, so it only reads World.
-unsafe impl<'w, 's, Q: ReadOnlyWorldQuery, F: ReadOnlyWorldQuery> ReadOnlySystemParam
-    for Query<'w, 's, Q, F>
+unsafe impl<'w, 's, Q: ReadOnlyWorldQuery + 'static, F: ReadOnlyWorldQuery + 'static>
```

Comment:
Was this required by the compiler? This seems incorrect to me.

Reply:
The `'static` bounds? They are required for the same reason I had to add them to `StaticSystemParam`. The `'static` bounds exist on the other impls for `Query`, so there's nothing lost here.

Reply:
oh awesome I didn't realise we were already requiring `Q: 'static` and `F: 'static` on the `SystemParam` impl. We should just require `trait WorldQuery: 'static` then

Reply:
Oh man that's nutty. Seems counterintuitive in this case, though probably for the best. LGTM then, we can address the `WorldQuery: 'static` thing in another PR.

Reply:
> We should just require trait WorldQuery: 'static then

I agree -- I just tried this though and it seems like it'll add a bunch of churn, so I'll leave it for another PR.